sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/StorageDirectoryDeletedEventData.java

```diff
@@ -153,7 +153,7 @@ public StorageDirectoryDeletedEventData setUrl(String url) {
      *
      * @return the recursive value.
      */
-    public Boolean isRecursive() {
+    public String getRecursive() {
```

Comment:
keep the method the same without changing the API, but change the method body to support the boolean.

Reply:
.net has the same approach.